src/AWS.Deploy.Common/Recipes/Validation/RecipeValidators/DockerfilePathValidator.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.IO;
+using AWS.Deploy.Common.IO;
+
+namespace AWS.Deploy.Common.Recipes.Validation
+{
+    /// <summary>
+    /// This validates that the Dockerfile is within the build context.
+    ///
+    /// Per https://docs.docker.com/engine/reference/commandline/build/#text-files
+    /// "The path must be to a file within the build context."
+    /// </summary>
+    public class DockerfilePathValidator : IRecipeValidator
+    {
+        public ValidationResult Validate(Recommendation recommendation, IDeployToolValidationContext deployValidationContext)
+        {
+            IDirectoryManager directoryManager = new DirectoryManager();
+
+            // Dockerfile uses the generic FilePath typehint, so we must load the latest value from the option setting.
+            // We also assume that if we detected a Dockerfile automatically it was set here via the replacement token
+            var dockerfilePath = recommendation.GetOptionSettingValue<string>(recommendation.GetOptionSetting("DockerfilePath"));
```

Comment:
I don't like how we are getting the DockerfilePath, we should be able to distinguish DeploymentBundle settings from regular Option settings

Reply:
Why don't you like this? Or why is it important to distinguish where the setting came from in the validator?

My understanding of here is that we're treating all OptionSettings the same, whether they came from the recipe or the deployment bundle.
https://github.com/aws/aws-dotnet-deploy/blob/f2e5db538a032be70de1dc3953b297528a62afeb/src/AWS.Deploy.Common/Recommendation.cs#L107

That same OptionSetting will be in this property, but it doesn't have the helper to find the one we're looking for.
https://github.com/aws/aws-dotnet-deploy/blob/f2e5db538a032be70de1dc3953b297528a62afeb/src/AWS.Deploy.Common/Recommendation.cs#L33

Reply:
We discussed this at the 5/5/22 internal meeting, switched to reading from `recommendation.DeploymentBundle` now that https://github.com/aws/aws-dotnet-deploy/pull/564 has been merged.